unit_tests/src/test_environment_variables.cpp

```diff
@@ -0,0 +1,92 @@
+
+#include <catch2/catch.hpp>
+
+#include "irods/getRodsEnv.h"
+#include "irods/irods_environment_properties.hpp"
+#include "irods/rodsErrorTable.h"
+
+#include <cstddef>
+#include <cstring>
+#include <string>
+
+// NOLINTNEXTLINE(readability-function-cognitive-complexity)
+TEST_CASE("environment_variables")
+{
+    const std::string env_var_key{"ISSUE_6866_ENV_KEY"};
+    const std::string env_var_val{"ISSUE_6866_ENV_VAL"};
+
+    irods::set_environment_property(env_var_key, env_var_val);
+
+    // Buffer size constants
+    constexpr std::size_t ZERO_BUFFER_SIZE = 0;
+    constexpr std::size_t SMALL_BUFFER_SIZE = 10;
```

Comment:
If this indicated that the buffer size is *too* small, that might be clearer (e.g. `TOOSMALL_BUFFER_SIZE`). Not required to change this since we are asserting that the size is smaller than the length of the value as part of the tests below - just a suggestion.